### DISCUSSION TITLE
How to read a pdf in Indian Languages

### DISCUSSION THREAD

**Discussion by cpumb** [pymupdf/PyMuPDF#1840] [Looking for help]
Hi,

I am trying to read a Kannada pdf book.

#Sample Code
pdf = fitz.open(text_book)
page = pdf.loadPage(24)  # number of page
print(page.get_text('text'))

When I read the file the output doesn't come properly.
“£ÀªÀÄä£ÀÄß ºÉZÀÄÑ ºÉZÀÄÑ zÀÄr¹PÉÆ¼ÀÄîwÛÃj. PÀrªÉÄ 
DºÁgÀ ¤ÃqÀÄwÛÃj. £ÀªÀÄä ¸ÉÃªÉ ¨ÉÃPÀÄ; £ÁªÀÅ ªÀiÁvÀæ 
¨ÉÃqÀ C®èªÉÃ? £ÀªÀÄä ¸ÉÃªÉAiÀÄ£ÀÄß ªÀÄgÉAiÀÄ¢j.”
“zÀAiÉÄ¬ÄgÀ° ¸ÀPÀ® ¥ÁætÂUÀ¼À°èè.”

Is there any way to set the language or font while extracting text?

**Comment by JorjMcKie** [comment]
This has nothing to do with the language of the text, but with the font used.
Not every font is prepared to support extraction of text. Many people are not aware of the fact, that **_displaying_** the text in a viewer and **_backtranslating_** it to unicode characters are two completely different things - and the second font feature (the so-called character map - CMAP) need not be present at all.
Encoding information is more often missing in Type 3 fonts - presumably your case.

If you want please share the PDF and I will take a look.
In the  meantime try and mark some text in a PDF viewer and copy / paste it into a word / office document and see what happens.

**Comment by cpumb** [comment]
[8th-language-kannada-1-without-watermark.pdf](https://github.com/pymupdf/PyMuPDF/files/9239763/8th-language-kannada-1-without-watermark.pdf)

Hi Jorj,
Attaching the PDF for your reference.

**Reply by JorjMcKie** [reply]
As suspected: this PDF contains fonts with non-standard encoding so none of the usual tools can extract the text.
